csharp/src/Ice/IceSSL/SSLEngine.cs

```diff
@@ -101,275 +98,137 @@ internal void initialize()
-                        e.reason = "IceSSL: no certificates found";
-                        throw e;
-                    }
+                    throw new Ice.InitializationException("IceSSL: no certificates found");
                 }
             }
 
-            if(_caCerts == null)
+            Debug.Assert(_caCerts == null);
+            string certAuthFile = properties.getProperty(prefix + "CAs");
+            if(certAuthFile.Length > 0 || properties.getPropertyAsInt(prefix + "UsePlatformCAs") <= 0)
             {
-                string certAuthFile = properties.getProperty(prefix + "CAs");
-                if(certAuthFile.Length == 0)
-                {
-                    certAuthFile = properties.getProperty(prefix + "CertAuthFile");
-                }
-                if(certAuthFile.Length > 0 || properties.getPropertyAsInt(prefix + "UsePlatformCAs") <= 0)
+                _caCerts = [];
+            }
+            if(certAuthFile.Length > 0)
+            {
+                if(!checkPath(ref certAuthFile))
                 {
-                    _caCerts = new X509Certificate2Collection();
+                    throw new Ice.InitializationException($"IceSSL: CA certificate file not found: {certAuthFile}");
                 }
-                if(certAuthFile.Length > 0)
+
+                try
                 {
-                    if(!checkPath(ref certAuthFile))
+                    using(FileStream fs = File.OpenRead(certAuthFile))
                     {
-                        Ice.PluginInitializationException e = new Ice.PluginInitializationException();
-                        e.reason = "IceSSL: CA certificate file not found: " + certAuthFile;
-                        throw e;
-                    }
+                        byte[] data = new byte[fs.Length];
+                        fs.Read(data, 0, data.Length);
 
-                    try
-                    {
-                        using(System.IO.FileStream fs = System.IO.File.OpenRead(certAuthFile))
+                        string strbuf = "";
+                        try
                         {
-                            byte[] data = new byte[fs.Length];
-                            fs.Read(data, 0, data.Length);
-
-                            string strbuf = "";
-                            try
-                            {
-                                strbuf = System.Text.Encoding.UTF8.GetString(data);
-                            }
-                            catch(Exception)
-                            {
-                                // Ignore
-                            }
+                            strbuf = System.Text.Encoding.UTF8.GetString(data);
+                        }
+                        catch(Exception)
+                        {
+                            // Ignore
+                        }
 
-                            if(strbuf.Length == data.Length)
+                        if(strbuf.Length == data.Length)
```

Comment:
Not sure, the PR doesn't update this code. Seems it try to guess if this a PEM certificate (which has only ASCII) text, otherwise for a DER the UTF8 conversion would fail and `strbuf.Length` would be 0.

I think we added this to allow multiple PEM certificates in a PEM encoded file, which other platforms support but C# didn't.